### PR TITLE
Implement OpenAI Swarm example with groupchat

### PR DESCRIPTION
**Pull request by yiranwu0**
Use groupchat's custom function to implement the openai swarm.

## Checks

- [ ] I've included any doc changes needed for https://autogenhub.github.io/autogen/. See https://autogenhub.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by yiranwu0** [comment]
Initial Commit. Todo: Organize code, draw stateflow figure and improve doc.

File: notebook/agentchat_swarm.ipynb [hunk 1]
```diff
@@ -0,0 +1,685 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Implement Swarm with AutoGen GroupChat\n",
+    "\n",
+    "\n",
+    "AutoGen offers conversable agents powered by LLM, tool or human, which can be used to perform tasks collectively via automated chat. Recently, OpenAI has released a [Swarm](https://github.com/openai/swarm) framework that focuses on making agent coordination and execution lightweight. In autogen, the groupchat allows customized speaker selection, which can be used to achieve the Swarm. This feature is also supported by our research paper [StateFlow: Enhancing LLM Task-Solving through State-Driven Workflows](https://autogen-ai.github.io/autogen/blog/2024/02/29/StateFlow).\n",
```
Comment (sonichi) [inline notebook/agentchat_swarm.ipynb]:
```suggestion
    "AutoGen offers conversable agents powered by LLM, tool or human, which can be used to perform tasks collectively via automated chat. Recently, OpenAI has released a [Swarm](https://github.com/openai/swarm) framework that focuses on making agent coordination and execution lightweight. In autogen, the groupchat allows customized speaker selection, which can be used to achieve the same orchestration pattern. This feature is also supported by our research paper [StateFlow: Enhancing LLM Task-Solving through State-Driven Workflows](https://autogen-ai.github.io/autogen/blog/2024/02/29/StateFlow).\n",
```

File: notebook/agentchat_swarm.ipynb [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+    "\n",
+    "In this notebook, we implements the [airline customer service example](https://github.com/openai/swarm/tree/main/examples/airline) from OpenAI Swarm.\n"
```
Comment (sonichi) [inline notebook/agentchat_swarm.ipynb]:
```suggestion
    "In this notebook, we implement the [airline customer service example](https://github.com/openai/swarm/tree/main/examples/airline) from OpenAI Swarm.\n"
```

**Review by sonichi — COMMENTED** [review]
Thanks. Could you add metadata to make it renderable on the website and searchable via tags?
After this PR is merged, in the next iteration, can we add some built-in speaker selection method so that users can directly import from the library to achieve the swarm-style orchestration?

**Comment by yiranwu0** [comment]
> Thanks. Could you add metadata to make it renderable on the website and searchable via tags? After this PR is merged, in the next iteration, can we add some built-in speaker selection method so that users can directly import from the library to achieve the swarm-style orchestration?

I added the tags to be compiled on the website. 
Yes, I will make another PR to implement this feature!

**Review by sonichi — APPROVED** [review]

**Pull request merged**